use-cases/vehicle-routing-time-windows/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutePlanResource.java

```diff
@@ -52,71 +62,129 @@ public VehicleRoutePlanResource(SolverManager<VehicleRoutePlan, String> solverMa
         this.solutionManager = solutionManager;
     }
 
+    @Operation(summary = "List the job IDs of all submitted route plans.")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "List of all job IDs.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(type = SchemaType.ARRAY, implementation = String.class))) })
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<String> list() {
+        return jobIdToJob.keySet().stream().toList();
```

Comment:
This has inefficiency built into it. The input is a set, the output is a list. Also, it is self-inflicted - nobody is forcing us to return a list.

And on top of that, it's done via a stream, which will do this inefficient thing inefficiently.